File: app/src/main/java/hexlet/code/games/EvenGame.java
package hexlet.code.games;

import java.util.Random;

public class EvenGame extends Game {

    private static final int MAX_NUMBER = 99;

    @Override
    public String getRule() {
        return "Answer 'yes' if the number is even, otherwise answer 'no'.";
    }

    @Override
    public GameRound getNewRound() {
        var number = new Random().nextInt(1, MAX_NUMBER + 1);

        return new GameRound(
                Integer.toString(number),
                rightAnswer(number)
        );
    }

    private String rightAnswer(int number) {
        return isEven(number) ? "yes" : "no";
    }

    private boolean isEven(int number) {
        return number % 2 == 0;
    }
}

File: app/src/main/kotlin/hexlet/code/EvenGame.kt
package hexlet.code

class EvenGame : Game() {
    override fun rule(): String {
        return "Answer 'yes' if the number is even, otherwise answer 'no'."
    }

    override fun newRound(): GameRound {
        val number = (1..99).random()
        return GameRound(question(number), rightAnswer(number))
    }

    private fun question(number: Int): String {
        return "Question: $number"
    }

    private fun rightAnswer(number: Int): String {
        return when {
            isEven(number) -> "yes"
            else -> "no"
        }
    }

    private fun isEven(number: Int): Boolean {
        return number % 2 == 0
    }
}

